pkg/teamloader/teamloader_test.go

```diff
@@ -128,3 +129,91 @@ func TestOverrideModel(t *testing.T) {
 		})
 	}
 }
+
+func openRoot(t *testing.T, dir string) *os.Root {
+	t.Helper()
+
+	root, err := os.OpenRoot(dir)
+	require.NoError(t, err)
+	t.Cleanup(func() { root.Close() })
+
+	return root
+}
+
+func TestCheckRequiredEnvVars(t *testing.T) {
+	tests := []struct {
+		yaml            string
+		expectedMissing []string
+	}{
+		{
+			yaml:            "openai_inline.yaml",
+			expectedMissing: []string{"OPENAI_API_KEY"},
+		},
+		{
+			yaml:            "anthropic_inline.yaml",
+			expectedMissing: []string{"ANTHROPIC_API_KEY"},
+		},
+		{
+			yaml:            "google_inline.yaml",
+			expectedMissing: []string{"GOOGLE_API_KEY"},
+		},
+		{
+			yaml:            "dmr_inline.yaml",
+			expectedMissing: []string{},
+		},
+		{
+			yaml:            "openai_model.yaml",
+			expectedMissing: []string{"OPENAI_API_KEY"},
+		},
+		{
+			yaml:            "anthropic_model.yaml",
+			expectedMissing: []string{"ANTHROPIC_API_KEY"},
+		},
+		{
+			yaml:            "google_model.yaml",
+			expectedMissing: []string{"GOOGLE_API_KEY"},
+		},
+		{
+			yaml:            "dmr_model.yaml",
+			expectedMissing: []string{},
+		},
+		{
+			yaml:            "all.yaml",
+			expectedMissing: []string{"ANTHROPIC_API_KEY", "GOOGLE_API_KEY", "OPENAI_API_KEY"},
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.yaml, func(t *testing.T) {
+			t.Parallel()
+
+			root := openRoot(t, "testdata/env")
+
+			cfg, err := config.LoadConfig(test.yaml, root)
```

Comment:
This is testing the config load, not the teamloader, that’s why I moved the code over there…